community/blog/2020.10-Introducing-message-handling-with-Azure-Service-Bus-message-influence/README.md

```diff
@@ -0,0 +1,121 @@
+            // Calling base operation where the Azure Service Bus operations are located.
+            await base.AbandonMessageAsync();
+        }
+        else
+        {
+            // Processing order.
+        }
+    }
+}
+```
+
+Note that in the regular message handling you don't have to specify which message to abandon. This is all done behind the screens so the message handler only has to focus on the message processing and call the specific Azure Service Bus message operations when necessary.
+The registration of this kind of message handler is just the same as any other regular message handler can can be added with the `.AddServiceBusMessageHandler` or `.AddMessageHandler` extensions.
+
+### Fallback message handling
+
+The fallback message handling is possible more correct when we're talking about dead lettering a message. Fallback message handlers will process the Azure Service Bus message when none other regular message handler was able to correctly process it.
+At the end of the fallback message processing, the message could be dead lettered on Azure for example.
+
+Here's an example of how a normal `IServiceBusFallbackMessageHandler` is implemented:
+
+```csharp
+public class OrderFallbackMessageHandler : IAzureServiceBusFallbackMessageHandler
+{
+    public async Task ProcessMessageAsync(Message orderMessage, AzureServiceBusMessageContext context, ...)
+    {
+        // Processing order.
+    }
+}
+```
+
+When you decide you want to dead letter the message, you'll have to implement  `AzureServiceBusFallbackMessageHandler` instead.
+
+```csharp
+public class OrderFallbackMessageHandler : AzureServiceFallbackMessageHandler
+{
+    public override async Task ProcessMessageAsync(Message orderMessage, AzureServiceBusMessageContext context, ...)
+    {
+        // Processing order.
+        await base.DeadLetterAsync(orderMessage);
+    }
+}
+```
+
+Note that because the fallback message handlers have more control over the original Azure Service Bus message, they will have to pass along the message to the base operations.
+
+Registering this fallback message handler is just the same as when implementing the interface:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddServiceBusMessageHandler<OrderCompletedMessageHandler>()
+            .AddServiceBusMessageHandler<OrderCancelledMessageHandler>()
+            .AddServiceBusFallbackMessageHandler<OrderFallbackMessageHandler>();
+}
+```
+
+## Is that all?
+
+No, of course not. The v0.4 and v0.5 version also has some minor updates regarding less verbose logging and some extra correlation extensions to pass along your own custom operation/transation ID property. All are helping in making the developer and consumer more in control of our messaging functionality.
+
+## What's on the horizon?
+
+We'll see how we can reuse our message handling system in the context of Azure functions where the function binding will replace the message pump.
```

Comment:
```suggestion
We are exploring how we can make our message handler approach available to customers using Azure Functions to provide a consistent experience. This would mean that Azure Functions would act as the message pump, but you can re-use your existing message handlers.
```